ql/pom.xml

```diff
@@ -32,6 +32,10 @@
     <!-- dependencies are always listed in sorted order by groupId, artifactId -->
     <!-- intra-project -->
     <!-- used for vector code-gen -->
+    <dependency>
+      <groupId>org.openjdk.nashorn</groupId>
```

Comment:
why do we need thiis?

Reply:
Nashorn Javascript engine was deprecated in jdk11 and was completely removed from jdk15 onwards.

Nashorn Javascript engine in hive is used for dynamic expression evaluation especially in partition filtering for temporary tables. it is added as an explicit dependency to compensate for its removal.
Sources: [https://openjdk.org/jeps/335](https://openjdk.org/jeps/335) , [https://openjdk.org/jeps/372](https://openjdk.org/jeps/372)

Reply:
got it, thanks!